backend/core/src/main/java/org/sonarsource/sonarlint/core/monitoring/MonitoringUserIdStore.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2025 SonarSource Sàrl
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.monitoring;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.sonarsource.sonarlint.core.UserPaths;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+
+public class MonitoringUserIdStore {
+
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+  private static final String USER_ID_FILE_NAME = "user_id.txt";
+
+  private final Path path;
+  @Nullable
+  private volatile UUID cachedUserId;
+
+  public MonitoringUserIdStore(UserPaths userPaths) {
+    this.path = userPaths.getUserHome().resolve(USER_ID_FILE_NAME);
```

Comment:
Everything we store, we put it under the 'storage root', I would advise to do the same and not store in the user home directly